src/modules/project/useCases/renseignerIdentifiantGestionnaireRéseau.factory.ts

```diff
@@ -0,0 +1,133 @@
+  | CodeEICNonTrouvéError
+  | IdentifiantGestionnaireRéseauObligatoireError
+  | IdentifiantGestionnaireRéseauExistantError
+>;
+
+export const renseignerIdentifiantGestionnaireRéseauFactory = ({
+  shouldUserAccessProject,
+  publish,
+  projectRepo,
+  trouverProjetsParIdentifiantGestionnaireRéseau,
+  gestionnaireRéseauRepo,
+}: Dependencies): CommandHandler => {
+  const vérifierCodeEIC = (command: Command) => {
+    if (command.codeEICGestionnaireRéseau) {
+      return gestionnaireRéseauRepo
+        .load(new UniqueEntityID(command.codeEICGestionnaireRéseau))
+        .andThen((gestionnaire) => {
+          if (gestionnaire.codeEIC !== command.codeEICGestionnaireRéseau) {
+            return errAsync(new CodeEICNonTrouvéError());
+          }
+          return okAsync(command);
+        });
+    }
+    return okAsync(command);
+  };
+
+  const chargerProjet = (command: { projetId: string; utilisateur: User }) => {
+    const { projetId, utilisateur } = command;
+    return wrapInfra(shouldUserAccessProject({ projectId: projetId, user: utilisateur })).andThen(
+      (utilisateurALesDroits) =>
+        utilisateurALesDroits
+          ? projectRepo.load(new UniqueEntityID(projetId)).map((projet) => ({
+              command,
+              projet,
+            }))
+          : errAsync(new UnauthorizedError()),
+    );
+  };
+
+  const vérifierIdentifiantGestionnaireRéseau = ({
+    command,
+    projet,
+  }: {
+    command: Command;
+    projet: Project;
+  }) => {
+    if (!command.identifiantGestionnaireRéseau) {
+      return errAsync(new IdentifiantGestionnaireRéseauObligatoireError());
+    }
+
+    return trouverProjetsParIdentifiantGestionnaireRéseau(
+      command.identifiantGestionnaireRéseau,
+    ).andThen((projets) => {
+      const existePourUnAutreProjet = projets.filter((p) => p !== command.projetId).length > 0;
+
+      return existePourUnAutreProjet
+        ? errAsync(new IdentifiantGestionnaireRéseauExistantError())
+        : okAsync({ command, projet });
+    });
+  };
+
+  const enregistrerLeChoix = ({
+    projet,
+    command: { projetId, utilisateur, identifiantGestionnaireRéseau, codeEICGestionnaireRéseau },
+  }: {
+    command: Command;
+    projet: Project;
+  }) =>
+    projet.identifiantGestionnaireRéseau !== identifiantGestionnaireRéseau
+      ? publish(
+          new NumeroGestionnaireSubmitted({
+            payload: {
+              projectId: projetId,
+              submittedBy: utilisateur.id,
+              numeroGestionnaire: identifiantGestionnaireRéseau,
+              ...(codeEICGestionnaireRéseau && { codeEICGestionnaireRéseau }),
+            },
+          }),
+        )
+      : codeEICGestionnaireRéseau
+      ? publish(
+          new GestionnaireRéseauRenseigné({
+            payload: {
+              projectId: projetId,
+              submittedBy: utilisateur.id,
+              codeEIC: codeEICGestionnaireRéseau,
+            },
+          }),
+        )
+      : okAsync(null);
```

Comment:
Je suis pas mega fan des double ternaires, je trouve que ça entrave la lisibilité du code. Je te propose l'implémentation suivante : 

```suggestion
  }) => {
    if (projet.identifiantGestionnaireRéseau !== identifiantGestionnaireRéseau) {
      return publish(
          new NumeroGestionnaireSubmitted({
            payload: {
              projectId: projetId,
              submittedBy: utilisateur.id,
              numeroGestionnaire: identifiantGestionnaireRéseau,
              ...(codeEICGestionnaireRéseau && { codeEICGestionnaireRéseau }),
            },
          }),
        )
    }

    if (codeEICGestionnaireRéseau) {
      return publish(
          new GestionnaireRéseauRenseigné({
            payload: {
              projectId: projetId,
              submittedBy: utilisateur.id,
              codeEIC: codeEICGestionnaireRéseau,
            },
          }),
        )
    }

    return okAsync(null)
  }
```